build_projects/shared-build-targets-utils/Utils/DotNetCli.cs

```diff
@@ -4,20 +4,22 @@
 using System.Runtime.InteropServices;
 using Microsoft.DotNet.Cli.Build.Framework;
 using Microsoft.DotNet.InternalAbstractions;
+using System.Collections.Generic;
 
 namespace Microsoft.DotNet.Cli.Build
 {
-    public class DotNetCli
+    public partial class DotNetCli
     {
         public static readonly DotNetCli Stage0 = new DotNetCli(GetStage0Path());
-        public static readonly DotNetCli Stage1 = new DotNetCli(Dirs.Stage1);
-        public static readonly DotNetCli Stage2 = new DotNetCli(Dirs.Stage2);
 
         public string BinPath { get; }
+        public IEnumerable<string> SharedFxVersionDirectories { get; private set; }
```

Comment:
Yeah that sounds good. I may have had a plan for it when I first made it but can't recall now. In general I agree, simpler is better.

---
In reply to: [66350264](https://github.com/dotnet/core-setup/pull/97#discussion_r66350264) [](ancestors = 66350264)